### PR TITLE
feat: Disable /notifications and /overrides by default

### PR DESCRIPTION
**Pull request by pawels-optimizely**
## Summary
- disable /notifications and /overrides by default
- allow them to be enabled via configuration

### CODE REVIEW

File: pkg/routers/api_v1_test.go [hunk 1]
```diff
@@ -72,6 +72,7 @@ func (suite *APIV1TestSuite) SetupTest() {
 		middleware:      &MockOptlyMiddleware{},
 		handlers:        MockHandlers{},
 		metricsRegistry: metricsRegistry,
+		enableOverrides: true,
```
Comment (mjc1283) [inline pkg/routers/api_v1_test.go]: I think we need a new test in this file, covering changes to `routers/api_v1.go`.

File: pkg/handlers/notification_disabled.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+// Package handlers //
+package handlers
+
+import (
+	"net/http"
+)
+
+// DisableNotificationHandler for disabled notifications
+type DisableNotificationHandler struct {
```
Comment (mikeng13) [inline pkg/handlers/notification_disabled.go]: nit: I'd name this `DisabledNotificationHandler` notice `Disable` vs `Disabled`

File: pkg/handlers/user_overrides_disabled.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+// Package handlers //
+package handlers
+
+import (
+	"net/http"
+)
+
+// DisableUserOverrideHandler implements the disabled overrides
+type DisableUserOverrideHandler struct{}
```
Comment (mikeng13) [inline pkg/handlers/user_overrides_disabled.go]: Same comment on `Disabled`

File: cmd/testdata/default.yaml [hunk 4]
```diff
@@ -24,6 +24,8 @@ admin:
 api:
   maxconns: 100
   port: "3000"
+  enablenotifications: true
+  enableoverrides: true
```
[the remark this answers is not in the document]
Comment (pawels-optimizely) [reply cmd/testdata/default.yaml]: > Suggest `enable_notifications`, `enable_overrides`

I know, I just tried to follow majority patterns, but it's still not consistent. We have a camel case: flushInterval or batchSize, all small case: sdkkeys or readtimeout. Introducing a snake case will not look nice. I think we need to clean up the definitions and go with either a camel case or snake case for all.

**Pull request merged**